aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -80,21 +80,51 @@ Tensor logdet(const Tensor& self) {
   return logdet_vals;
 }
 
-std::tuple<Tensor, Tensor> slogdet(const Tensor& self) {
+std::tuple<Tensor, Tensor> linalg_slogdet(const Tensor& self) {
   squareCheckInputs(self);
-  TORCH_CHECK((at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type())),
-              "Expected a floating point tensor as input");
+  ScalarType t = self.scalar_type();
+  TORCH_CHECK((t == ScalarType::Double || t == ScalarType::Float || t == ScalarType::ComplexFloat || t == ScalarType::ComplexDouble)
+              && self.dim() >= 2,
+              "linalg_slogdet(", self.scalar_type(), "{", self.sizes(), "}): expected a tensor with 2 or more dimensions "
+              "of float, double, cfloat or cdouble types");
 
   Tensor det_P, diag_U;
   std::tie(det_P, diag_U) = _lu_det_P_diag_U(self);
-  auto det_sign = diag_U.sign().prod(-1).mul_(det_P);
+  auto det_sign = diag_U.sgn().prod(-1).mul_(det_P);
   // abslogdet_val is -inf if U is singular, in which case diag_U.abs_().log_().sum(-1) will return -inf.
   // U is singular when U(i, i) = 0 for some i in [1, self.size(-1)].
   // Since abslogdet_val cannot take nan, no special case handling is required.
-  auto abslogdet_val = diag_U.abs_().log_().sum(-1);
+  // in-place abs is not supported for complex tensors
+  auto abslogdet_val = diag_U.abs().log_().sum(-1);
   return std::make_tuple(det_sign, abslogdet_val);
 }
 
+// TODO: implement _out variant avoiding copy and using already allocated storage directly
+std::tuple<Tensor&, Tensor&> linalg_slogdet_out(const Tensor& input, Tensor& sign, Tensor& logabsdet) {
+  TORCH_CHECK(sign.scalar_type() == input.scalar_type(),
+    "sign dtype ", sign.scalar_type(), " does not match input dtype ", input.scalar_type());
+  ScalarType real_dtype = toValueType(typeMetaToScalarType(input.dtype()));
+  TORCH_CHECK(logabsdet.scalar_type() == real_dtype,
+    "logabsdet dtype ", logabsdet.scalar_type(), " does not match the expected dtype ", real_dtype);
+  TORCH_CHECK(sign.device() == input.device() && logabsdet.device() == input.device(),
+              "Expected sign, logabsdet and input to be on the same device, but found sign on ",
+              sign.device(), ", logabsdet on ", logabsdet.device(), " and input on ", input.device(), " instead.");
+
+  Tensor sign_tmp, logabsdet_tmp;
+  std::tie(sign_tmp, logabsdet_tmp) = at::linalg_slogdet(input);
+
+  at::native::resize_output(sign, sign_tmp.sizes());
+  sign.copy_(sign_tmp);
+  at::native::resize_output(logabsdet, logabsdet_tmp.sizes());
+  logabsdet.copy_(logabsdet_tmp);
+
+  return std::tuple<Tensor&, Tensor&>(sign, logabsdet);
+}
+
+std::tuple<Tensor, Tensor> slogdet(const Tensor& self) {
```

Comment:
So with this change `torch.slogdet` is an alias for `torch.linalg.slogdet`? 

Although they might diverge if we change the namedtuple names to match NumPy.

Reply:
Yes, now `torch.slogdet` is an alias for `torch.linalg.slogdet` so it inherits for example complex support from it. Is there a special way to register aliases?

Reply:
Kind of, but we don't need to worry about it at this time.

Would you file an issue, however, that torch.slogdet should be made an alias of torch.linalg.slogdet? Then we can follow-up on that (low priority) issue at a later tdate.

Aliasing is a pattern more than a registry, although there is a registry in the jit that remaps aliases, there's no actual value from remapping svd to linalg.svd in torchscript. That remapping is intended for more common operations, like chunk or split or div, which torchscript might be able to optimize.